crates/cxx-qt-lib/src/core/qdate.rs

```diff
@@ -72,11 +146,74 @@ impl fmt::Debug for QDate {
     }
 }
 
+impl From<i64> for QDate {
+    /// Converts the Julian day jd to a QDate.
+    fn from(jd: i64) -> Self {
+        Self { jd }
+    }
+}
+
+impl From<&QDate> for i64 {
+    /// Converts the date to a Julian day.
+    fn from(date: &QDate) -> i64 {
+        date.jd
+    }
+}
+
 impl QDate {
+    /// Returns a QDate object containing a date ndays later than the date of this object (or earlier if ndays is negative).
+    ///
+    /// Returns a null date if the current date is invalid or the new date is out of range.
+    pub fn add_days(&self, ndays: i64) -> Self {
+        ffi::qdate_add_days(self, ndays)
+    }
+
+    // Returns the current date, as reported by the system clock.
+    pub fn current_date() -> Self {
+        ffi::qdate_current_date()
+    }
+
+    /// Returns the number of days from this date to d (which is negative if d is earlier than this date).
+    ///
+    /// Returns 0 if either date is invalid.
+    pub fn days_to(&self, date: Self) -> i64 {
+        ffi::qdate_days_to(self, date)
+    }
+
+    /// Returns the time as a string. The format parameter determines the format of the result string.
+    pub fn format(&self, format: &ffi::QString) -> ffi::QString {
+        ffi::qdate_to_format(self, format)
+    }
+
+    /// Converts the Julian day jd to a QDate.
+    pub fn from_julian_day(jd: i64) -> Self {
+        Self { jd }
+    }
+
+    /// Returns the QTime represented by the string, using the format given, or an invalid time if the string cannot be parsed.
```

Comment:
So the `from_string` allows for custom formats https://doc.qt.io/qt-6/qdate.html#fromString-4 eg `QDate::fromString("1MM12car2003", "d'MM'MMcaryyyy");`. Whereas `from_string_enum` is limited to a few known formats https://doc.qt.io/qt-6/qt.html#DateFormat-enum

Reply:
Ah, alright, I thought it was using the string to select a DateFormat. My bad.

Then I guess this can stay this way, with the Option<Self> return value.
Still not a great fan of the `_enum` suffix, but I don't have any better ideas either...